### PR TITLE
Merge shelf packages

### PR DESCRIPTION
**Pull request by natebosch**
Merge dart-lang/shelf_* packages into this repo.

For each package, including `shelf`, create a commit within it's source
repo that moves code into a `pkgs` subdirectory. Remove the github
actions workflows that would conflict and will be overridden by
`mono_repo generate`.

Create a merge commit for each package into `shelf` to retain history.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# See https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-actions-up-to-date-with-dependabot
+
+version: 2
+updates:
+
+- package-ecosystem: "github-actions"
+  directory: "/"
+  schedule:
+    # Check for updates to GitHub Actions every weekday
+    interval: "daily"
```
Comment (kevmoo) [inline .github/dependabot.yml:11]: monthly? I find all of the updates a bit crazy...

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -0,0 +1,1071 @@
+# Created with package:mono_repo v6.0.0
```
Comment (kevmoo) [inline .github/workflows/dart.yml:1]: You should grab the latest mono_repo!

File: .github/dependabot.yml (hunk 1 above)
Comment (natebosch) [reply .github/dependabot.yml:11]: This one came from one of the other repos, I'm not actually sure if dependabot works on mono repos...

Comment (kevmoo) [reply .github/dependabot.yml:11]: It does for actions – but you'll be limited by what mono_repo has setup

Comment (natebosch) [reply .github/dependabot.yml:11]: hmm, maybe should just drop it then.

Comment (kevmoo) [reply .github/dependabot.yml]: so far I've only seen this as noise – especially for monorepos that use mono_repo

File: .test_config [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "test_package": {
+    "platforms": ["vm"]
+  }
+}
```
Comment (kevmoo) [inline .test_config:5]: ??? Do we even care about this file any more?

Comment (natebosch) [reply .test_config:5]: nope, must have missed this in one of the repos.

**Comment by natebosch** [comment]
Plan is to merge this to master with a fast-forward merge.

I have rewritten all the tags from the outside repositories to have a prefix of the package name, and a consistent `v` before the version number. The tags were not perfect before - there may be some missing and at least one where the version number is incomplete - but I am not planning on doing further manual cleanup. It is safe to rename tags from the other repositories, because in the places where they have meaning that must be immutable it is also associated with the specific repository. After merging this PR I will push the tags as well. The tags that originated in this repository will remain unchanged, and going forward we'll add package name prefixes to all new tags including for `shelf`.

```
0.5.7
0.6.0
0.6.1
0.6.1+1
0.6.1+2
0.6.1+3
0.6.2
0.6.2+1
0.6.3
0.6.3+1
0.6.4
0.6.4+1
0.6.4+2
0.6.4+3
0.6.5
0.6.5+1
0.6.5+2
0.6.5+3
0.6.6
0.6.7
0.6.7+1
0.6.7+2
0.6.8
0.7.0
0.7.1
0.7.2
0.7.3
0.7.3+1
0.7.3+2
0.7.3+3
0.7.4
0.7.4+1
0.7.5
0.7.6
0.7.7
0.7.8
0.7.9
1.0.0
1.0.0-nullsafety.0
1.1.0
1.1.1
1.1.2
1.1.3
1.1.4
1.2.0
1.3.0
shelf_packages_handler-v1.0.0
shelf_packages_handler-v1.0.1
shelf_packages_handler-v1.0.2
shelf_packages_handler-v1.0.3
shelf_packages_handler-v1.0.4
shelf_packages_handler-v2.0.0
shelf_packages_handler-v2.0.1
shelf_packages_handler-v3.0.0
shelf_proxy-v0.1.0+2
shelf_proxy-v0.1.0+3
shelf_proxy-v0.1.0+4
shelf_proxy-v0.1.0+6
shelf_proxy-v0.1.0+7
shelf_proxy-v1.0.0
shelf_proxy-v1.0.1
shelf_static-v0.1.0
shelf_static-v0.1.1
shelf_static-v0.1.1+1
shelf_static-v0.1.4+2
shelf_static-v0.1.4+3
shelf_static-v0.1.4+4
shelf_static-v0.1.4+5
shelf_static-v0.2.0
shelf_static-v0.2.1
shelf_static-v0.2.2
shelf_static-v0.2.3+1
shelf_static-v0.2.3+2
shelf_static-v0.2.3+3
shelf_static-v0.2.3+4
shelf_static-v0.2.4
shelf_static-v0.2.5
shelf_static-v0.2.6
shelf_static-v0.2.7
shelf_static-v0.2.7+1
shelf_static-v0.2.8
shelf_static-v0.2.9+1
shelf_static-v0.2.9+2
shelf_static-v1.0.0
shelf_static-v1.1
shelf_test_handler-v1.0.0
shelf_test_handler-v1.0.1
shelf_test_handler-v1.0.2
shelf_test_handler-v1.0.3
shelf_test_handler-v2.0.0
shelf_web_socket-v0.0.1+2
shelf_web_socket-v0.0.1+3
shelf_web_socket-v0.0.1+4
shelf_web_socket-v0.0.1+5
shelf_web_socket-v0.1.0
shelf_web_socket-v0.2.0
shelf_web_socket-v0.2.1
shelf_web_socket-v0.2.2
shelf_web_socket-v0.2.2+1
shelf_web_socket-v0.2.2+2
shelf_web_socket-v0.2.2+3
shelf_web_socket-v0.2.2+4
shelf_web_socket-v0.2.2+5
shelf_web_socket-v0.2.3
shelf_web_socket-v0.2.4
shelf_web_socket-v0.2.4+1
shelf_web_socket-v1.0.0
shelf_web_socket-v1.0.1
```

**Review by kevmoo — APPROVED** [review]
#yolo 😎

**Comment by natebosch** [comment]
Hmm, @kevmoo I think we need to do the merge outside of github for this and I can't push to master.

**Pull request merged**